test/src/test/java/org/corfudb/integration/AddNodeIT.java

```diff
@@ -0,0 +1,145 @@
+                .runServer();
+
+        CorfuRuntime n1Rt = new CorfuRuntime(getConnectionString(n1Port)).connect();
+
+        SMRMap<String, String> map = n1Rt.getObjectsView()
+                .build()
+                .setType(SMRMap.class)
+                .setStreamName(streamName)
+                .open();
+
+        final int numEntris = 20_000;
+        for (int x = 0; x < numEntris; x++) {
+            map.put(String.valueOf(x), String.valueOf(x));
+
+            if (x % 200 == 0) {
+                System.out.println("iter " + x);
+            }
+        }
+
+        // Add a second node
+        final int n2Port = 9001;
+        new CorfuServerRunner()
+                .setHost(host)
+                .setPort(n2Port)
+                .runServer();
+
+        ManagementClient mgmt = n1Rt.getRouter(getConnectionString(n1Port))
+                .getClient(ManagementClient.class);
+
+        OrchestratorResponse orchResp = mgmt.addNodeRequest(getConnectionString(n2Port)).get();
+        AddNodeResponse resp = (AddNodeResponse) orchResp.getResponse();
+        UUID workflowId = resp.getWorkflowId();
+
+        boolean status = false;
+        int maxTries = 10;
+
+        for (int x = 0; x < maxTries; x++) {
+            try {
+                orchResp = mgmt.queryRequest(workflowId).get();
+                status = ((QueryResponse) orchResp.getResponse()).isActive();
+                System.out.println("Status for " + workflowId + " " + status);
+                Thread.sleep(2000);
+            } catch (ExecutionException e) {
+                if (e.getCause() instanceof WrongEpochException) {
+                    n1Rt.invalidateLayout();
+                } else {
+                    throw e;
+                }
+            }
+        }
+
+        n1Rt.invalidateLayout();
+        assertThat(n1Rt.getLayoutView().getLayout().getAllServers().size()).isEqualTo(2);
+
+        MultiCheckpointWriter mcw = new MultiCheckpointWriter();
+        mcw.addMap(map);
+
+        long prefix = mcw.appendCheckpoints(n1Rt, "Maithem");
+
+        n1Rt.getAddressSpaceView().prefixTrim(prefix - 1);
+
+        n1Rt.getAddressSpaceView().invalidateClientCache();
+        n1Rt.getAddressSpaceView().invalidateServerCaches();
+        n1Rt.getAddressSpaceView().gc();
+
+        // Add a 3rd node after compaction
+        final int n3Port = 9002;
+        new CorfuServerRunner()
+                .setHost(host)
+                .setPort(n3Port)
+                .runServer();
+
+        orchResp = mgmt.addNodeRequest(getConnectionString(n3Port)).get();
+        resp = (AddNodeResponse) orchResp.getResponse();
+        workflowId = resp.getWorkflowId();
+
+        for (int x = 0; x < maxTries; x++) {
+            try {
+                orchResp = mgmt.queryRequest(workflowId).get();
+                status = ((QueryResponse) orchResp.getResponse()).isActive();
+                System.out.println("Status for " + workflowId + " " + status);
```

Comment:
what is the for loop trying to do ? Is it just blindly retrying with sleep. I could not see any other condition check ?